sdk/eventhub/Azure.Messaging.EventHubs/src/Primitives/EventProcessor.cs

```diff
@@ -2027,12 +2032,18 @@ private async Task ValidateStorageConnectionAsync(CancellationToken cancellation
 
         /// <summary>
         ///   Calculates the maximum number of partitions that it is advised this processor
-        ///   own, based on the host environment.
+        ///   own, based on the host environment and some general heuristics.
         /// </summary>
         ///
         /// <returns>The maximum number of partitions that it is advised this processor own.</returns>
         ///
-        private static int CalculateMaximumAdvisedOwnedPartitions() => (int)Math.Floor(Environment.ProcessorCount * 1.5);
+        /// <remarks>
+        ///   The safe number of partitions owned will vary quite a bit by application logic, event structure, and the
+        ///   host environment.  This is a general approximation based on support issues observed since the initial
+        ///   processor release.
+        /// </remarks>
+        ///
+        private static int CalculateMaximumAdvisedOwnedPartitions() => (Environment.ProcessorCount * 2);
```

Comment:
Did I read incorrectly, or do the docs say this should be 3?
```suggestion
        private static int CalculateMaximumAdvisedOwnedPartitions() => (Environment.ProcessorCount * 3);
```

Reply:
It's intentionally 2 here.   The maximum that we recommend is 3, but I want to issue a warning at 2 that there may be problems.